### DISCUSSION TITLE
Extract Constant

### DISCUSSION THREAD

**Discussion by dciborow** [microsoft/pylance-release#2915] [Enhancement]
Currently VSCode features the "Extract Variable" command. "Extract Constant", would behave similarly but...
1. Extracted Variable is moved to top of file 
2. Variable is given name with ALL_CAPS
3. Any other instances of the variable could optional also be replaced, prompted by a pop up "Replace X other occurs of this?", which provides options to "Find", "Replace", or "Replace All".

Python Code Before Running Extraction

```
def func1(x="Constant"):
  print(x)


def func2(x=None):
  x = x or "Constant"
  return x

print("Here is another usage of Constant")
```

After
```
CONSTANT = "Constant"

def func1(x=CONSTANT ):
  print(x)


def func2(x=None):
  x = x or CONSTANT 
  return x

print(f"Here is another usage of {CONSTANT}")
```

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by cmurga-pal** [comment]
Is this ever gona be implemented?

**Reply by rchiodo** [reply]
We do things generally based on up votes. You can see the sorting like so:

https://github.com/microsoft/pylance-release/discussions?discussions_q=is%3Aopen+sort%3Atop

**Reply by rchiodo** [reply]
If a discussion item is open, it means we haven't added it to our backlog yet.

**Reply by cmurga-pal** [reply]
Thanks for the quick response!
